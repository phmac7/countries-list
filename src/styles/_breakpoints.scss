// ========================================
// BREAKPOINTS
// ========================================

// Design Breakpoints (Frontend Mentor Style Guide)
$mobile-width: 520px;
$desktop-width: 1440px;

// Responsive Breakpoints (for development)
$breakpoint-mobile: 520px;
$breakpoint-tablet: 768px;
$breakpoint-desktop: 1440px;

// ========================================
// MEDIA QUERY MIXINS
// ========================================

@mixin mobile {
  @media (max-width: $breakpoint-mobile) {
    @content;
  }
}

@mixin tablet {
  @media (min-width: $breakpoint-tablet) {
    @content;
  }
}

@mixin desktop {
  @media (min-width: $breakpoint-desktop) {
    @content;
  }
}

@mixin tablet-only {
  @media (min-width: $breakpoint-tablet) and (max-width: #{$breakpoint-desktop - 1px}) {
    @content;
  }
}
