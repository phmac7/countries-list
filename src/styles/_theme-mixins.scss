// Mapa de cores por tema
$themes: (
  light: (
    text-color: #111517,
    background-color: #fafafa,
    element-color: #ffffff,
    border-color: #e1e5e9,
    shadow-color: rgba(0, 0, 0, 0.1),
    hover-color: #f5f5f5,
    input-color: #b2b2b2,
  ),
  dark: (
    text-color: #ffffff,
    background-color: #202c37,
    element-color: #2b3743,
    border-color: #3a4a5c,
    shadow-color: rgba(0, 0, 0, 0.3),
    hover-color: #374151,
    input-color: #fff,
  ),
);

// Função para obter cor do tema
@function theme-color($color-name) {
  @return var(--#{$color-name});
}

// Mixin para aplicar tema
@mixin theme($property, $color-name) {
  #{$property}: theme-color($color-name);
}

// Mixin para criar variáveis CSS
@mixin generate-theme-variables {
  :root {
    @each $theme-name, $theme-colors in $themes {
      @if $theme-name == 'light' {
        @each $color-name, $color-value in $theme-colors {
          --#{$color-name}: #{$color-value};
        }
      }
    }
  }

  @each $theme-name, $theme-colors in $themes {
    @if $theme-name != 'light' {
      [data-theme='#{$theme-name}'] {
        @each $color-name, $color-value in $theme-colors {
          --#{$color-name}: #{$color-value};
        }
      }
    }
  }
}
