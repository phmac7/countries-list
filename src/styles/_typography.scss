@use 'variables' as *;

h1 {
  font-size: 3.2rem;
  font-weight: $font-weight-extrabold;
  line-height: 1.2;
}

h2 {
  font-size: 2.4rem;
  font-weight: $font-weight-extrabold;
  line-height: 1.3;
}

h3 {
  font-size: 2rem;
  font-weight: $font-weight-semibold;
  line-height: 1.4;
}

h4 {
  font-size: 1.8rem;
  font-weight: $font-weight-semibold;
  line-height: 1.4;
}

h5 {
  font-size: 1.6rem;
  font-weight: $font-weight-semibold;
  line-height: 1.4;
}

h6 {
  font-size: 1.4rem;
  font-weight: $font-weight-semibold;
  line-height: 1.4;
}

article,
aside,
nav,
section {
  h1 {
    font-size: 2.4rem;
  }
}
