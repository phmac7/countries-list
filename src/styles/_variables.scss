//Colors
$blue-900: #2b3743; // Dark Mode Elements
$blue-950: #202c37; // Dark Mode Background
$grey-950: #111517; // Light Mode Text
$grey-400: #858585; // Light Mode Input
$grey-50: #fafafa; // Light Mode Background
$white: #ffffff; // Dark Mode Text & Light Mode Elements

// Typography

$font-family-base: var(--font-nunito-sans), 'Nunito Sans', sans-serif;
$font-weight-light: 300;
$font-weight-semibold: 600;
$font-weight-extrabold: 800;

$font-size-homepage: 1.4rem;
$font-size-detail: 1.6rem;

// Transitions

$transition-base: all 0.2s ease-in-out;
$transition-fast: all 0.1s ease-in-out;
$transition-slow: all 0.3s ease-in-out;
