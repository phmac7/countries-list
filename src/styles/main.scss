@use 'variables' as *;
@use 'breakpoints' as *;
@use 'theme-mixins' as *;
@use 'typography' as *;

@include generate-theme-variables;

[data-theme='light'] {
  body {
    color: theme-color(text-color);
    background-color: theme-color(background-color);
  }
}

[data-theme='dark'] {
  body {
    color: theme-color(text-color);
    background-color: theme-color(background-color);
  }
}

html {
  font-size: 62.5%;
}

html,
body {
  max-width: 100vw;
  overflow-x: hidden;
}

body {
  font-family: $font-family-base;
  font-size: $font-size-detail;
  font-weight: $font-weight-light;
  line-height: 1.5;
  color: theme-color(text-color);
  background-color: theme-color(background-color);
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

a {
  color: inherit;
  text-decoration: none;
}

ul {
  list-style: none;
}
