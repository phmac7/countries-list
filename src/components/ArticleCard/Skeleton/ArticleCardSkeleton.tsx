'use client';
import React from 'react';
import styles from './ArticleCardSkeleton.module.scss';

export const ArticleCardSkeleton = () => {
  return (
    <div className={styles.skeleton} data-testid="article-card-skeleton">
      <div className={styles.skeleton__image} data-testid="skeleton-image" />
      <div className={styles.skeleton__content} data-testid="skeleton-content">
        <div className={styles.skeleton__title} data-testid="skeleton-title" />
        <div className={styles.skeleton__text} data-testid="skeleton-text" />
        <div className={styles.skeleton__text} data-testid="skeleton-text" />
        <div className={styles.skeleton__text} data-testid="skeleton-text" />
      </div>
    </div>
  );
};
