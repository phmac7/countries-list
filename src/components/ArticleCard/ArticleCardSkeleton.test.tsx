import { render, screen } from '@testing-library/react';
import { ArticleCardSkeleton } from './ArticleCardSkeleton';
import styles from './ArticleCardSkeleton.module.scss';

describe('ArticleCardSkeleton', () => {
  it('should render with correct test id', () => {
    render(<ArticleCardSkeleton />);
    expect(screen.getByTestId('article-card-skeleton')).toBeInTheDocument();
  });

  it('should render all skeleton elements', () => {
    render(<ArticleCardSkeleton />);

    expect(screen.getByTestId('skeleton-image')).toBeInTheDocument();
    expect(screen.getByTestId('skeleton-content')).toBeInTheDocument();
    expect(screen.getByTestId('skeleton-title')).toBeInTheDocument();

    const textPlaceholders = screen.getAllByTestId('skeleton-text');
    expect(textPlaceholders).toHaveLength(3);
  });

  it('should have proper structure', () => {
    const { container } = render(<ArticleCardSkeleton />);
    expect(container.querySelector('.skeleton')).toMatchSnapshot();
  });

  it('should have correct image dimensions', () => {
    render(<ArticleCardSkeleton />);
    const image = screen.getByTestId('skeleton-image');
    expect(image).toHaveClass(styles.skeleton__image);
  });
});
