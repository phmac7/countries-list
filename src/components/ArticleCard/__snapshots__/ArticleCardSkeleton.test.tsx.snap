// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ArticleCardSkeleton should have proper structure 1`] = `
<div
  class="skeleton"
  data-testid="article-card-skeleton"
>
  <div
    class="skeleton__image"
    data-testid="skeleton-image"
  />
  <div
    class="skeleton__content"
    data-testid="skeleton-content"
  >
    <div
      class="skeleton__title"
      data-testid="skeleton-title"
    />
    <div
      class="skeleton__text"
      data-testid="skeleton-text"
    />
    <div
      class="skeleton__text"
      data-testid="skeleton-text"
    />
    <div
      class="skeleton__text"
      data-testid="skeleton-text"
    />
  </div>
</div>
`;
